File: compiler/DOC/BYTECODE/coroutines/beta-based/Coroutine.java
/*
   compilation:
     javac -classpath "$BETALIB/basiclib/jvm;$BETALIB/basiclib/private/jvm;." Coroutine.java
 */

import beta.BetaObject;
import beta.Component;

public abstract class Coroutine extends BetaObject {
   
    public Coroutine(){
	comp$ = new Component(this);
    }

    public void call() { comp$.swap(); }

    public void suspend() { comp$.swap(); }
}

File: compiler/DOC/BYTECODE/coroutines/beta-based/Coroutine.cs
/*
   compilation:
     csc -nologo -t:exe -r:"$BETALIB/basiclib/clr/betaenv.dll,$BETALIB/basiclib/private/clr/Component.dll" Coroutine.cs
 */

public abstract class Coroutine: BetaObject {
   
    public Coroutine(){
	_comp = new Component(this);
    }

    public void call() { _comp.swap(); }

    protected void suspend() { _comp.swap(); }
}
